modules/fundamental/src/purl/service/mod.rs

```diff
@@ -230,6 +230,30 @@ impl PurlService {
         }
     }
 
+    pub async fn purls_by_purl<C: ConnectionTrait>(
+        &self,
+        purls: Vec<Purl>,
+        deprecation: Deprecation,
+        connection: &C,
+    ) -> Result<Vec<PurlDetails>, Error> {
+        let canonical: Vec<CanonicalPurl> = purls
+            .iter()
+            .map(|purl| CanonicalPurl::from(purl.clone()))
+            .collect();
+
+        let items = qualified_purl::Entity::find()
+            .filter(qualified_purl::Column::Purl.is_in(canonical))
+            .all(connection)
+            .await?;
+
+        let mut details = Vec::with_capacity(items.len());
+        for purl in items {
+            details
+                .push(PurlDetails::from_entity(None, None, &purl, deprecation, connection).await?);
+        }
+        Ok(details)
+    }
```

Comment:
I don't love that both this fn and `purl_by_purl` exist. They're mostly redundant. I'd say we can probably delete `purl_by_purl` in favor of this one, and any current caller of `purl_by_purl` can just grab the first element returned by this one. Make sense?

Reply:
Yes, perfect sense.